teuthology/run_tasks.py

```diff
@@ -111,10 +111,11 @@ def run_tasks(tasks, ctx):
                 'task': taskname,
                 'owner': ctx.owner,
             }
-            if 'teuthology_branch' in config:
-                tags['teuthology_branch'] = config['teuthology_branch']
-            if 'branch' in config:
-                tags['branch'] = config['branch']
+            optional_tags = ('teuthology_branch', 'branch', 'suite',
+                             'machine_type', 'os_type', 'os_version')
+            for tag in optional_tags:
+                if tag in config:
+                    tags[tag] = config[tag]
```

Comment:
Why do not use standard teuthology deep_merge and invent something new?

Reply:
it's doing something different - we're only adding a subset of values from the config. deep_merge would include all of them.

Reply:
yeah... I see it now